### PR TITLE
Fix edge case in TokenList.replace when match is at file boundary

### PR DESCRIPTION
**Pull request by BLannoo**
Addresses an edge case where token replacement produced incorrect results when the matched range fell at the very start or end of the token list.

Reproduces with inputs where the original text begins or ends the file content without surrounding context tokens.

### CODE REVIEW

File: agents/agents-ext/src/jvmTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolJvmTest.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        val result = edit(
+            p,
+            "return a + b",
+            "return a + b + 0 // no-op"
+        )
+
+        assertTrue(result.applied)
+        assertTrue(p.readText().contains("return a + b + 0 // no-op"))
+    }
+}
```
Comment (mytestingaccount2026) [inline agents/agents-ext/src/jvmTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolJvmTest.kt:198]: **Missing trailing newline — breaks CI.**

The ktlint check fails on this file: `File must end with a newline (\n)`. This is the direct cause of the `compilation` CI failure on this PR.

File: agents/agents-ext/src/commonTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolMultilineContentTest.kt [hunk 2]
```diff
@@ -1,143 +0,0 @@
-package ai.koog.agents.ext.tool.file
```
Comment (mytestingaccount2026) [inline agents/agents-ext/src/commonTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolMultilineContentTest.kt:1]: **Significant test coverage lost — including the boundary cases the PR title claims to address.**

This file covered several scenarios that have no equivalent in the new `EditFileToolJvmTest`:

- `test_start_of_file_replacement_via_tool` — start-of-file boundary
- `test_end_of_file_replacement_via_tool` — end-of-file boundary
- `test_consecutive_edits_via_tool` — sequential edits on the same file
- `test_large_content_middle_replacement_via_tool` — large file (2000+ lines)

The first two are especially notable since the PR title specifically mentions "match is at file boundary". These should be ported to the new test class.

File: agents/agents-ext/src/commonTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolFormattingEdgeCasesTest.kt [hunk 3]
```diff
@@ -1,75 +0,0 @@
-package ai.koog.agents.ext.tool.file
```
Comment (mytestingaccount2026) [inline agents/agents-ext/src/commonTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolFormattingEdgeCasesTest.kt:1]: **Formatting edge case coverage dropped without replacement.**

These two tests are not ported to the new JVM test class:

- `test_tabs_vs_spaces_equivalent_original_is_applied` — verifies tab/space normalization during fuzzy matching
- `test_crlf_vs_lf_equivalent_original_is_applied` — verifies line-ending normalization

These are valuable edge cases (especially CRLF/LF which matters for cross-platform use). Consider adding equivalents to `EditFileToolJvmTest`.

File: agents/agents-ext/src/jvmTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolJvmTest.kt [hunk 4]
```diff
@@ -0,0 +1,198 @@
+package ai.koog.agents.ext.tool.file
```
Comment (mytestingaccount2026) [inline agents/agents-ext/src/jvmTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolJvmTest.kt:1]: **Tests moved from `commonTest` to `jvmTest` — loses multiplatform coverage.**

The deleted tests lived in `commonTest` and ran on all Kotlin targets. This new file is JVM-only. For a Kotlin Multiplatform project, this means `EditFileTool` behavior is no longer validated on non-JVM targets.

If the intent is to test against the real filesystem (via `JVMFileSystemProvider` + `@TempDir`), that's reasonable for a JVM integration test — but the `InMemoryFS`-based unit tests in `commonTest` served a complementary purpose and didn't need to be removed to add these.

**Review by mytestingaccount2026 — CHANGES_REQUESTED** [review]
**PR title and description are misleading.** The title claims "Fix edge case in TokenList.replace when match is at file boundary" and the description talks about fixing incorrect token replacement results — but there are **zero production code changes**. The actual content is a test restructuring: deleting 4 `commonTest` files and adding 1 `jvmTest` file. The commit messages (":fire: remove tests for experimenting with test generation" and "Add JVM tests for EditFileTool") tell the real story. Please update the PR title and description to reflect what this PR actually does.

**Summary of issues:**

1. **CI is broken** — the new file is missing a trailing newline, failing ktlint.
2. **Test coverage regression** — ~8 test scenarios from the deleted files have no equivalent in the new test class, including boundary conditions, tab/space normalization, CRLF/LF handling, delete operations, large files, consecutive edits, and different file types.
3. **Multiplatform coverage lost** — tests moved from `commonTest` (all targets) to `jvmTest` (JVM only). The `InMemoryFS`-based common tests and the new real-FS JVM tests serve complementary purposes; both should coexist.
